MapboxTileWriter coordinates count check

kolyanch opened the pull request:
Found two problems during testing.
* For polygons, the encoding method inserts an extra point to close the rings. This should be done only via the ClosePath command, otherwise the last point is duplicated after reading the tile.
Added coordinates count check in RoundTrip geometry check method.

* After adding a check for the number of coordinates, the multipoint test stopped working because some points were discarded due to negative coordinates at this line:
https://github.com/NetTopologySuite/NetTopologySuite.IO.VectorTiles/blob/83c856de17e447634b94ac9bb321b9040ce0bce3/src/NetTopologySuite.IO.VectorTiles.Mapbox/MapboxTileWriter.cs#L215

This check looks strange. I didn't quite understand what was being checked here. 
Why does one coordinate of a point have to be positive? 
A point can still go beyond the extent's boundaries if one coordinate is negative or larger than the extent size (4096).

Removed this check for now, but I'm not sure it's right)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/NetTopologySuite/NetTopologySuite.IO.VectorTiles?pullRequest=21) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/NetTopologySuite/NetTopologySuite.IO.VectorTiles?pullRequest=21) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/NetTopologySuite/NetTopologySuite.IO.VectorTiles?pullRequest=21) it.</sub>

FObermaier commented:
@kolyanch, thanks for your contribution. 
Adding points that are not within the extent of a vector tile consumes unneccessary space, which is not desirable in this format.
I think that's why the points with negative coordinates are discarded.

FObermaier commented:
Nice catch! Regarding the `MultiPoint` handling, could you add your example to the `RoundTripTests`, then I'll merge it.

kolyanch commented:
In the last commit, I returned the check, added two tests for multipoints.
Skipping a point is not so easy. I had to save the previous point in order to correctly process the relative shifts for the following points. Fixed the ```CheckGeometry``` method for the correct comparison of geometries for ```MultiPoint```.
